### PR TITLE
SF and logic change

### PR DESCRIPTION
**Pull request by neilking**
sf and code logic change. I don't think breaking up UserLanguageDetectorMessageListener into methods is needed, since the logic is straight forward and not complex in this project. We can break up into method unless the logic become more complex.

### CODE REVIEW

**Comment by samuelkong** [comment]
Still not done:
https://github.com/samuelkong/liferay-sdg/pull/28#discussion_r51514627

**Comment by samuelkong** [comment]
> > It doesn't need to follow Brian standard, but at least run source-formatter
>
> Doesn't look like this was done

Better, but it's still not done. Running format-source doesn't mean just running it. It means also fixing the problems that it reports.

**Comment by samuelkong** [comment]
> I don't think breaking up UserLanguageDetectorMessageListener into methods is needed, since the logic is straight forward and not complex in this project. We can break up into method unless the logic become more complex. 

You have this huge comment block in your code to explain the logic and you don't think it's complex?

File: languagedetector/language-detector/src/main/java/com/liferay/languagedetector/messaging/UserLanguageDetectorMessageListener.java [hunk 1]
```diff
@@ -47,19 +46,20 @@ public void receive(Message message) throws MessageListenerException {
...
+			 * time is null, language is zh_CN, -> add time stamp
+			 * time is null, language is not zh_CN - > return
 			 * (no need to implement)
-			 * time is not null, language is not en, more/equal than 10,
-			 * set user languate to be en_US, remove timestamp from expando.
+			 * time is not null, language is not zh_CN -> remove time stamp
+			 * time is not null, language is zh_CN, less than 20, return
+			 * (no need to implement)
+			 * time is not null, language is zh_CN, more/equal than 20 min,
+			 * set user language to be en_US, remove timestamp from expando.
```
Comment (samuelkong) [inline languagedetector/language-detector/src/main/java/com/liferay/languagedetector/messaging/UserLanguageDetectorMessageListener.java]: Language should not be changed to en_US.

From Skype:
> No, I mean if the user is not on Chinese then change it to Chinese after 20 minutes of not being on Chinese

**Comment by samuelkong** [comment]
I'm changing the requirement again:

1. Instead of `zh_CN`, I want `de_DE` now.
2. Instead of 20 minutes, I want 17 minutes now.

**Comment by samuelkong** [comment]
Merged into master

**Comment by neilking** [comment]
"I'm changing the requirement again:
Instead of zh_CN, I want de_DE now.
Instead of 20 minutes, I want 17 minutes now."

You have changed requirement 2 times. I feel you want to do something else rather than changing requirement only. Please clarify your requirement, and make it be straight forward.

Here are some idea you may want
1. changing language and time without code change.
2. changing language and time through UI.

I won't keep working on it unless you clarify your requirement or fix the requirement.

File: languagedetector/language-detector/src/main/java/com/liferay/languagedetector/messaging/UserLanguageDetectorMessageListener.java (hunk 1 above)
Comment (neilking) [inline languagedetector/language-detector/src/main/java/com/liferay/languagedetector/messaging/UserLanguageDetectorMessageListener.java]: I can remove the comment. I still don't think it's necessary.

**Comment by samuelkong** [comment]
>  2. changing language and time through UI.

I don't need this.

> I won't keep working on it unless you clarify your requirement or fix the requirement.

But why is this so hard for you? These should be simple changes that take you less then 1 minute to make. If it takes you more then 1 minute, you're doing something wrong.

**Comment by neilking** [comment]
Do you mean you want to change the language through properties file?

**Comment by samuelkong** [comment]
Not necessary. The point is you should be writing code such that these type of changes should be super fast.

**Comment by neilking** [comment]
Make these type of changes super fast means it's some sense of configurable, 3 ways I can think of change it fast. 1 local static of dynamic variable(require code change and redeployment), 2 properties file, (no code change but require redeployment), 3, ui configuration,(no code change and redeployment). What do you want?

**Comment by samuelkong** [comment]
Either 1 or 2 is fine. And you should have done this from the very beginning. This is not a Liferay thing. This is just standard best practices when you write code and I expect you to follow best practices when writing any code.

3 is not necessary unless there's a client requirement for this. And in this case I've said this is not necessary.

**Pull request closed without merging**